graphprot/prepare_graphprot_seqs.py

```diff
@@ -0,0 +1,274 @@
+    # otherwise request manual definition of chromosome limits
+    if (len(chromsizes) == 0):
+        logging.error("Error: retrieving chromosome sizes from UCSC failed. Please specify manually using parameter --chromosome_limits")
+        exit(1)
+
+# output file arguments
+pos_core_bed_fn = args.output_file_prefix + ".positives_core.bed"
+neg_core_bed_fn = args.output_file_prefix + ".negatives_core.bed"
+# TODO: use
+pos_seq_bed_fn = args.output_file_prefix + ".positives_seq.bed"
+# TODO: use
+neg_seq_bed_fn = args.output_file_prefix + ".negatives_seq.bed"
+pos_seq_fa_fn = args.output_file_prefix + ".positives.fa"
+neg_seq_fa_fn = args.output_file_prefix + ".negatives.fa"
+
+# calculate flank lengths
+flank_length = args.seq_length - args.core_length
+flank_upstream_length = int(flank_length / 2)
+flank_downstream_length = int(flank_length / 2) + (flank_length % 2)
+if (args.core_length + flank_upstream_length + flank_downstream_length != args.seq_length):
+    raise Exception("Error: bad length calculation.")
+
+
+def dbg_head(sites, description="", n=npeek, run=args.debug):
+    """Print the first few bed entries."""
+    if (run):
+        logging.debug(description)
+        for i in sites[0:n]:
+            logging.debug(i)
+
+
+def prefix_neg(feature, prefix="negative_from_"):
+    """Modify BedTool feature by adding a prefix."""
+    feature.name = prefix + feature.name
+    return feature
+
+
+def offset_zero_by_one(feature):
+    """Sets the start coordinate to 1 if it is actually 0.
+
+    Required for the flanking to work properly in those cases.
+    """
+    if (feature.start == 0):
+        feature.start = feature.start + 1
+    return feature
+
+
+def get_flanks(cores,
+               flank_upstream_length=flank_upstream_length,
+               flank_downstream_length=flank_downstream_length):
+    """Calculate flanking regions of a core region."""
+    if (args.chromosome_limits is not None):
+        logging.debug("using chromosome_limits " + args.chromosome_limits)
+        # get upstream flanks
+        flanks_upstream = cores.flank(
+            s=True,
+            l=flank_upstream_length,
+            r=0,
+            g=args.chromosome_limits).saveas()
+        # get downstream flanks
+        flanks_downstream = cores.flank(
+            s=True,
+            r=flank_downstream_length,
+            l=0,
+            g=args.chromosome_limits).saveas()
+    else:
+        # get upstream flanks
+        flanks_upstream = cores.flank(
+            s=True,
+            l=flank_upstream_length,
+            r=0,
+            genome=args.genome_id).saveas()
+        # get downstream flanks
+        flanks_downstream = cores.flank(
+            s=True,
+            r=flank_downstream_length,
+            l=0,
+            genome=args.genome_id).saveas()
+    # check if sites and flanks have the same number of entries
+    if (not cores.count() == flanks_upstream.count() == flanks_downstream.count()):
```

Comment:
This looks wired. You want to have the not infront of the brackets? Is working, but I think this syntax is not really educative.